packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -26,15 +29,20 @@ import 'input_date_picker_form_field.dart';
 import 'input_decorator.dart';
 import 'material.dart';
 import 'material_localizations.dart';
+import 'material_state.dart';
 import 'scaffold.dart';
 import 'text_button.dart';
 import 'text_field.dart';
 import 'text_theme.dart';
 import 'theme.dart';
 
-const Size _calendarPortraitDialogSize = Size(330.0, 518.0);
+// The M3 sizes are coming from the tokens, but are hand coded,
+// as the current token DB does not contain landscape versions.
```

Comment:
Could you put these values into the template so that if they add the landscape versions we only need to change the template?

Reply:
The way I handled this for the time picker was to have an intermediary abstract class for the defaults that had some local size values that could be used in the code, but weren't exposed in the public Theme API. For example, [here](https://cs.opensource.google/flutter/flutter/+/master:packages/flutter/lib/src/material/time_picker.dart;l=3129;drc=59767e5fc2b85232fe7fe6c29ed13a144105b878). They could then be overridden for M2 and M3 variants.

The other advantage of that approach was that all of the properties can be change to return non-nullable versions, so that none of the references in the code need to have a `!` on them.